src/beatwave/level.cpp

```diff
@@ -14,12 +17,50 @@
 
 Level::Level(const char *levelFilePath)
 {
+    pugi::xml_document doc;
+
+    doc.load_file(levelFilePath);
+
+    m_player.reset(new Player(doc.child("player")));
```

Comment:
If think the corresponding classes should check that and throw an exception if something goes wrong.